Base: [Reajuste do salário mínimo em 2018 é o menor em 24 anos](https://g1.globo.com/economia/noticia/reajuste-do-salario-minimo-em-2018-e-o-menor-em-24-anos.ghtml)

Segundo notícia publicada na globo.com:

> O reajuste de 1,81% no valor do salário mínimo para o ano de 2018, de R$ 937 para R$ 954, é o menor desde o Plano Real, anunciado em 1994 e que controlou a hiperinflação na economia brasileira, aponta série histórica do Departamento Intersindical de Estatística e Estudos Socioeconômicos (Dieese).

In [1]:
require 'httparty'
require 'nokogiri'

true

Coleta os dados do salário mínimo a partir da tabela contida na página do Departamento Intersindical de Estatística e Estudos Socioeconômicos:

In [104]:
url = 'https://www.dieese.org.br/analisecestabasica/salarioMinimo.html'
page = HTTParty.get(url)
parsed_page = Nokogiri::HTML(page)
tabela = parsed_page.css('table')

nil

Extrai os salários da tabela agrupados por ano e os armazena em um dicionário:

In [90]:
linhas = tabela.css('tr')

ano_salarios = {}
ano = nil

linhas.each do |linha|
  numerico = linha.text.to_i

    if numerico != 0
      ano = numerico
      ano_salarios[numerico] = []
    end
      
    if numerico == 0
      if ano_salarios[ano] != nil
        colunas = linha.css('td')
        
        minimo_nominal = colunas[1].text
        minimo_nominal.gsub!('R$ ', '').gsub!(',', '.')
        ano_salarios[ano] << minimo_nominal.to_f
      end
    end
end

0

Quais os salários registrados no ano de 2017?

In [91]:
ano_salarios[2017]

[937.0, 937.0, 937.0, 937.0, 937.0, 937.0, 937.0, 937.0, 937.0, 937.0, 937.0]

Quais os salários registrados no ano de 1998?

In [92]:
ano_salarios[1998]

[130.0, 130.0, 130.0, 130.0, 130.0, 130.0, 130.0, 130.0, 120.0, 120.0, 120.0, 120.0]

Calcula a média anual de salários por ano:

In [102]:
ano_salario = {}

ano_salarios.each do |ano, salarios|
  ano_salario[ano] = salarios.sum / salarios.length
end

nil

Qual o salário médio em 2017?

In [96]:
puts ano_salario[2017]

937.0


Valor anunciado por Temer para 2018 é de R$ 954,00

In [221]:
ano_salario[2018] = 954.0

954.0

In [222]:
menor_ano = ano_salario.keys.min

1994

In [223]:
reajuste = {}

ano_salario.each do |ano, salario|
  if ano > menor_ano
    diferenca = salario - ano_salario[ano-1]
    reajuste[ano] = diferenca / ano_salario[ano - 1] * 100
  end
end

{2017=>937.0, 2016=>880.0, 2015=>788.0, 2014=>724.0, 2013=>678.0, 2012=>622.0, 2011=>544.1666666666666, 2010=>510.0, 2009=>460.8333333333333, 2008=>409.1666666666667, 2007=>372.5, 2006=>337.5, 2005=>286.6666666666667, 2004=>253.33333333333334, 2003=>230.0, 2002=>195.0, 2001=>172.75, 2000=>147.25, 1999=>134.0, 1998=>126.66666666666667, 1997=>117.33333333333333, 1996=>108.0, 1995=>90.0, 1994=>68.26333333333334, 2018=>954.0}

In [224]:
reajuste[2018]

1.814300960512273

In [225]:
puts "O salário mínimo de R$ #{ano_salario[2018]} apresenta"\
" um reajuste de #{reajuste[2018].round(2)}% em relação ao ano"\
" anterior"

O salário mínimo de R$ 954.0 apresenta um reajuste de 1.81% em relação ao ano anterior


Ordena de forma crescente todos os reajustes armazenados:

In [196]:
reajuste_ordenado = reajuste.sort_by { |ano, reajuste| reajuste }.to_h

{2018=>1.814300960512273, 1999=>5.789473684210522, 2017=>6.4772727272727275, 2011=>6.699346405228751, 2014=>6.784660766961652, 1998=>7.954545454545463, 1997=>8.641975308641971, 2015=>8.83977900552486, 2013=>9.003215434083602, 2008=>9.8434004474273, 2000=>9.888059701492537, 2004=>10.144927536231888, 2007=>10.37037037037037, 2010=>10.669077757685356, 2016=>11.6751269035533, 2009=>12.627291242362515, 2002=>12.879884225759769, 2005=>13.157894736842108, 2012=>14.303215926493118, 2001=>17.31748726655348, 2006=>17.732558139534877, 2003=>17.94871794871795, 1996=>20.0, 1995=>31.842375115972455}

Ao listar todos os reajustes ordenados, é possível perceber que o reajuste de 2018 (1.81%) é o menor reajuste em 24 anos

(ou seja, desde a anunciação do Plano Real, em 1994).

O menor aumento até então corresponde ao abservado em 1999, (5,79%).

In [219]:
reajuste_ordenado.each_with_index do |(ano, reajuste), index|
  posicao = index + 1
  puts "#{posicao}: #{reajuste.round(2)}% (#{ano})"
end

1: 1.81% (2018)
2: 5.79% (1999)
3: 6.48% (2017)
4: 6.7% (2011)
5: 6.78% (2014)
6: 7.95% (1998)
7: 8.64% (1997)
8: 8.84% (2015)
9: 9.0% (2013)
10: 9.84% (2008)
11: 9.89% (2000)
12: 10.14% (2004)
13: 10.37% (2007)
14: 10.67% (2010)
15: 11.68% (2016)
16: 12.63% (2009)
17: 12.88% (2002)
18: 13.16% (2005)
19: 14.3% (2012)
20: 17.32% (2001)
21: 17.73% (2006)
22: 17.95% (2003)
23: 20.0% (1996)
24: 31.84% (1995)


{2018=>1.814300960512273, 1999=>5.789473684210522, 2017=>6.4772727272727275, 2011=>6.699346405228751, 2014=>6.784660766961652, 1998=>7.954545454545463, 1997=>8.641975308641971, 2015=>8.83977900552486, 2013=>9.003215434083602, 2008=>9.8434004474273, 2000=>9.888059701492537, 2004=>10.144927536231888, 2007=>10.37037037037037, 2010=>10.669077757685356, 2016=>11.6751269035533, 2009=>12.627291242362515, 2002=>12.879884225759769, 2005=>13.157894736842108, 2012=>14.303215926493118, 2001=>17.31748726655348, 2006=>17.732558139534877, 2003=>17.94871794871795, 1996=>20.0, 1995=>31.842375115972455}